File: backend/src/main/java/com/xxgwy/common/response/PageResult.java
package com.xxgwy.common.response;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class PageResult<T> {
    private int code;
    private String message;
    private List<T> data;
    private long total;
    private int page;
    private int pageSize;

    public static <T> PageResult<T> of(List<T> data, long total, int page, int pageSize) {
        return new PageResult<>(200, "ok", data, total, page, pageSize);
    }
}

File: backend/src/main/kotlin/com/xxgwy/common/response/PageResult.kt
package com.xxgwy.common.response

data class PageResult<T>(
    val list: List<T>,
    val total: Long,
    val page: Int,
    val size: Int
) {
    companion object {
        @JvmStatic
        fun <T> of(list: List<T>, total: Long, page: Int, size: Int): PageResult<T> =
            PageResult(list, total, page, size)
    }
}
